### PR TITLE
Bash completion for flags

### PR DESCRIPTION
**Pull request by mabelzhang**
# 🎉 New feature

Part of https://github.com/gazebosim/gz-tools/issues/1
Used together with https://github.com/gazebosim/gz-tools/pull/87

## Summary

Bash completion, installation, tests.

## Test it

```
$ . install/share/gz/gz.completion

$ ign gazebo -
--force-version         --physics-engine        --verbose
--gui-config            --playback              --version
--help                  --record                --versions
--iterations            --record-path           -g
--levels                --record-resources      -h
--log-compress          --record-topic          -r
--log-overwrite         --render-engine         -s
--network-role          --render-engine-gui     -v
--network-secondaries   --render-engine-server  -z

$ ign model -
--force-version  --list           -h               -p
--help           --model          -j               
--joint          --pose           -l               
--link           --versions       -m
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/cmd/gazebo.bash_completion.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
```
Comment (ahcorde) [inline src/cmd/gazebo.bash_completion.sh:1]: LICENSE

File: src/cmd/gazebo.bash_completion.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# This is a per-library function definition, used in conjunction with the
+# top-level entry point in ign-tools.
+
+# NOTE: In Garden+, replace `gazebo` in function name to align with subcommand,
+# for ign-tools/etc/ign.bash_completion.sh to find this function.
+function _gz_gazebo
+{
+  if [[ ${COMP_WORDS[COMP_CWORD]} == -* ]]; then
+    # Specify options (-*) word list for this subcommand
```
Comment (ahcorde) [inline src/cmd/gazebo.bash_completion.sh:13]: Add a TODO to include `--headless-rendering` from Fortress

File: src/ign_TEST.cc [hunk 3]
```diff
@@ -160,3 +161,68 @@ TEST(CmdLine, ResourcePath)
...
+  EXPECT_NE(std::string::npos, output.find("--render-engine-server")) << output;
+  EXPECT_NE(std::string::npos, output.find("--version")) << output;
+  EXPECT_NE(std::string::npos, output.find("-z")) << output;
+  EXPECT_NE(std::string::npos, output.find("--help")) << output;
+  EXPECT_NE(std::string::npos, output.find("--force-version")) << output;
+  EXPECT_NE(std::string::npos, output.find("--versions")) << output;
+
+  // Flags in bash completion
+  std::ifstream scriptFile(std::string(PROJECT_SOURCE_PATH) +
+    "/src/cmd/gazebo.bash_completion.sh");
```
Comment (ahcorde) [inline src/ign_TEST.cc:199]: `ignition::common::jointPaths`

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Review by ahcorde — CHANGES_REQUESTED** [review]
CI is failing to compile
```
     std::string scriptPath = common::joinPaths(std::string(PROJECT_SOURCE_PATH),
```

**Comment by mabelzhang** [comment]
Oops my Docker image was broken yesterday and I couldn't test things locally. Relying on CI

**Comment by mabelzhang** [comment]
[b34bbe9](https://github.com/gazebosim/gz-sim/pull/1504/commits/b34bbe9ed7246c77706c10773427acd397d4c920) adds bash completion for the `model` command and tests it in a new way as described in https://github.com/gazebosim/gz-gui/pull/392#discussion_r889390294

File: src/cmd/CMakeLists.txt [hunk 4]
```diff
@@ -117,3 +117,26 @@ set(ign_model_ruby_path "${cmd_model_script_generated_test}")
 configure_file(
   "model.yaml.in"
   "${CMAKE_BINARY_DIR}/test/conf/model${PROJECT_VERSION_MAJOR}.yaml" @ONLY)
+
+
```
Comment (ahcorde) [inline src/cmd/CMakeLists.txt:121]:
```suggestion
```

**Review by chapulina — APPROVED** [review]
Pushed some minor tweaks in 7c0f3b9bcffce2a7b9b3feca7eb897c1d6ff0011. LGTM!

**Pull request merged**